### PR TITLE
Move f64::NAN ui tests into `library`

### PR DESCRIPTION
**Pull request by chansuke**
This is a partial fix of #76268.

r? @matklad

### CODE REVIEW

**Comment by m-ou-se** [comment]
The automated code formatting check is failing because of the 2-space indentation. Can you apply the formatting suggested by rustfmt (4-space indentation)? `./x.py fmt` will do that for you.

File: library/core/tests/num/nan.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#[test]
+fn test_nan() {
+  use core::f64;
+  let x = "NaN".to_string();
+  assert_eq!(format!("{}", f64::NAN), x);
+  assert_eq!(format!("{:e}", f64::NAN), x);
+  assert_eq!(format!("{:E}", f64::NAN), x);
```
Comment (jyn514) [inline library/core/tests/num/nan.rs:7]:
```suggestion
    use core::f64;
    let x = "NaN".to_string();
    assert_eq!(format!("{}", f64::NAN), x);
    assert_eq!(format!("{:e}", f64::NAN), x);
    assert_eq!(format!("{:E}", f64::NAN), x);
```

**Review by jyn514 — COMMENTED** [review]
r=me with the whitespace fixed

**Comment by jyn514** [comment]
Thanks for the PR!

**Comment by ollie27** [comment]
Will this test actually be run? Doesn't it need `mod nan;` adding to `library\core\tests\num\mod.rs`?

**Comment by jyn514** [comment]
Oops, good catch!

@bors r-

**Pull request merged**